identity-android/src/main/java/com/android/identity/android/direct_access/DirectAccess.kt

```diff
@@ -0,0 +1,507 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.android.direct_access
+
+import com.android.identity.cbor.Bstr
+import com.android.identity.cbor.Cbor
+import com.android.identity.cbor.Cbor.decode
+import com.android.identity.crypto.X509Cert
+import com.android.identity.crypto.X509CertChain
+import com.android.identity.direct_access.DirectAccessTransport
+import com.android.identity.util.Logger
+import org.bouncycastle.asn1.ASN1UTCTime
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.security.cert.CertificateException
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.text.SimpleDateFormat
+import java.util.Calendar
+
+/**
+ * A class which handles common operations related to the DirectAccess applet.
+ *
+ * @param transport the method of interacting with the applet.
+ */
+class DirectAccess(val transport: DirectAccessTransport) {
+
+    companion object {
+        private const val TAG = "DirectAccessCredential"
+
+        val DIRECT_ACCESS_PROVISIONING_APPLET_ID =
+            byteArrayOf(0xA0.toByte(), 0x00, 0x00, 0x02, 0x48, 0x00, 0x01, 0x01, 0x01)
+
+        private const val PROVISION_BEGIN: Byte = 0
+        private const val PROVISION_UPDATE: Byte = 1
+        private const val PROVISION_FINISH: Byte = 2
+        private const val KEY_CERT: Int = 0x01
+        private const val KEY_ENC_DATA: Int = 0x00
+        private const val CMD_MDOC_CREATE = 0x01
+        private const val CMD_MDOC_CREATE_PRESENTATION_PKG = 0x07
+        private const val CMD_MDOC_DELETE_CREDENTIAL = 0x08
+        private const val CMD_MDOC_PROVISION_DATA = 0x09
+        private const val CMD_MDOC_SWAP_IN = 0x06
+        private const val APDU_RESPONSE_STATUS_OK = 0x9000
+        private const val INS_ENVELOPE = 0xC3.toByte()
+    }
+
+    /**
+     * Returns the maximum size, KiB, of `credentialData` which can be used in credentials. This
+     * will be at least 32 KiB.
+     */
+    val maximumCredentialSize: Long
+        get() {
+            // todo integrate with applet
+            return 32
+        }
```

Comment:
Feels weird this is in KiB, suggest to just return number of bytes (e.g. 32768 instead) otherwise someone is just going to get the conversion wrong. (And if we wanted to convey it in kibibytes, the variable should have `KiB` or `Kibibytes` as a suffix.)